clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -65,6 +65,9 @@
     /* The v2 metadata request is the same as v1. An additional field for cluster id has been added to the v2 metadata response */
     public static final Schema METADATA_REQUEST_V2 = METADATA_REQUEST_V1;
 
+    /* The v3 metadata request is the same as v1 and v2. An additional field for throttle time has been added to the v3 metadata response */
```

Comment:
Discussed this with @junrao. The main challenge with this option is having the top level error field for every response. This would probably affect a lot of code:

1. We would need to handle this top level error code everywhere.
2. A bunch of protocols that currently have a top level error code would no longer have them, so a bunch of code would have to be updated as well.

So, it doesn't seem appropriate to do this as part of this KIP.